src/derivative_utils.jl

```diff
@@ -129,28 +129,30 @@ If `integrator.f` has a custom Jacobian update function, then it will be called.
 either automatic or finite differencing will be used depending on the `cache`.
 """
 function calc_J!(J, integrator, cache)
-  @unpack t,uprev,f,p,alg = integrator
+  @unpack dt,t,uprev,f,p,alg = integrator
 
+  tdt = t + dt
```

Comment:
This makes sense for always_new where we want the Jacobian at `t + dt`. Do we want this in general? @ChrisRackauckas

Reply:
I guess Rosenbrock methods might not like this.

Reply:
But it's supposed to be evaluated at `t` if it's at `uprev`?

Reply:
Yes, I guess we should only do that when it's always_new.